src/app/shop/[category]/[productId]/_components/Product/OptionWithButtons.tsx

```diff
@@ -0,0 +1,116 @@
+'use client';
+
+import { postCart } from '@/api/cartAPI';
+import { Button, Dropdown } from '@/components';
+import type { CartProductType, OptionTypes } from '@/types/ProductTypes';
+import { useMutation } from '@tanstack/react-query';
+import classNames from 'classnames/bind';
+import { useState } from 'react';
+import OptionContainer from './OptionContainer';
+import styles from './ProductDetail.module.scss';
+import QuantitySelector from './QuantitySelector';
+
+const cn = classNames.bind(styles);
+
+interface OptionWithButtonProps {
+  productId: number;
+  optionList: OptionTypes[];
+  price: number;
+}
+interface SelectedOptionType {
+  id: number;
+  name: string;
+  count: number;
+}
+
+const OPTION_PLACEHOLDER = '스위치 (필수)';
+
+export default function OptionWithButton({ productId, optionList, price }: OptionWithButtonProps) {
+  const optionNames = optionList?.map((option) => option.optionName);
+  const [selectedOptions, setSelectedOptions] = useState<SelectedOptionType[]>([]);
+  const totalPrice = selectedOptions.reduce((acc, option) => acc + option.count * price, 0);
+  const [noOptionCount, setNoOptionCount] = useState<number>(1);
+
+  const handleChangeOption = (value: string) => {
+    if (value !== OPTION_PLACEHOLDER) {
+      const selectedOption = optionList.find((option) => option.optionName === value);
+      if (selectedOption) {
+        const { id, optionName } = selectedOption;
+
+        setSelectedOptions((prevOptions) => {
+          const isExisting = prevOptions.find((option) => option.name === value);
+          if (isExisting) {
+            return prevOptions.map((option) =>
+              option.name === optionName ? { ...option, count: option.count + 1 } : option,
+            );
+          } else {
+            return [...prevOptions, { id, name: optionName, count: 1 }];
+          }
+        });
+      }
+    }
+  };
+
+  const handleUpdateCount = (name: string, count: number) => {
+    setSelectedOptions((prevOptions) =>
+      prevOptions.map((option) => (option.name === name ? { ...option, count } : option)),
+    );
+  };
+
+  const handleDeleteOption = (name: string) => {
+    setSelectedOptions((prevOptions) => prevOptions.filter((option) => option.name !== name));
+  };
+
+  const addCartProduct = useMutation({ mutationFn: (data: CartProductType) => postCart(data) });
```

Comment:
```suggestion
  const { mutate: addCartProduct } = useMutation({ mutationFn: (data: CartProductType) => postCart(data) });
```
여기에서는 mutate만 사용되어서 괜찮은데 useMutataion의 다른 기능들도 사용하실땐 이렇게 사용하는게 더 편하실거에요